addons/stock/report/report_deliveryslip.xml

```diff
@@ -129,7 +129,8 @@
                                 </t>
                                 <!-- Make sure we do another section for package-less products if they exist -->
                                 <t t-set="move_lines" t-value="o.move_line_ids.filtered(lambda l: not l.result_package_id)"/>
-                                <t t-if="move_lines" name="no_package_move_lines">
+                                <t t-set="empty_moves" t-value="o.move_line_ids._get_empty_moves()"/>
+                                <t t-if="move_lines or empty_moves" name="no_package_move_lines">
```

Comment:
I think you can do that:
```suggestion
                                <t t-set="aggregated_lines" t-value="move_lines._get_aggregated_product_quantities() or o.move_line_ids._get_empty_moves()"/>
                                <t t-if="move_lines or aggregated_lines" name="no_package_move_lines">
```
... and then, you can remove the lines you're adding (line 145 to line 150) and also replace the lines from 140 to 142 by:
```xml
                                    <t t-elif="aggregated_lines">
                                        <t t-call="stock.stock_report_delivery_aggregated_move_lines"/>
                                    </t>
```

Reply:
Thanks !